src/Core/ServiceProvider.php

```diff
@@ -40,15 +40,15 @@ abstract class ServiceProvider extends AbstractServiceProvider implements Bootab
      * This method's actual return type and documented return type differ
      * because PHP 7.2 doesn't support return type narrowing.
      *
-     * @return \Cake\Core\ContainerInterface
+     * @return \League\Container\DefinitionContainerInterface
      */
-    public function getContainer(): PsrContainerInterface
+    public function getContainer(): DefinitionContainerInterface
```

Comment:
I think we will need to create our own `DefinitionContainerInterface` which extends `DefinitionContainerInterface`.

Reply:
Safer in the end, allows certain shimming where needed.

Reply:
Agreed. We could start off with a simple extension and add if we need to. Having our own interface helps keep league container out of user land code more.